### PR TITLE
pool: improve struct m0_pool_version doc

### PR DESCRIPTION
**Pull request by andriytk**
Extend struct m0_pool_version documentation with a few more clarifying sentences.

### CODE REVIEW

File: pool/pool.h [hunk 1]
```diff
@@ -98,7 +98,14 @@ struct m0_pool {
 };
 
 /**
- * Pool version is the subset of devices from the filesystem.
+ * Pool version is the subset of devices from the filesystem
+ * configuration: racks, controllers, disks, etc. If represent
+ * the whole cluster hardware configuration as a tree, the pool
+ * version would be its sub-tree.
```
Comment (huanghua78) [inline pool/pool.h:104]: If the whole cluster hardware configuration is represented as a tree, the pool version would be its sub-tree.

File: pool/pool.h [hunk 2]
```diff
@@ -98,7 +98,14 @@ struct m0_pool {
...
+ *
+ * Mero objects and indexes are assigned some pool version when
```
Comment (huanghua78) [inline pool/pool.h:106]: s/Mero/Motr

File: pool/pool.h (hunk 1 above)
Comment (andriytk) [reply pool/pool.h:104]: Accepted, thanks.

File: pool/pool.h (hunk 2 above)
Comment (andriytk) [reply pool/pool.h:106]: Right, fixed. :)

**Review by huanghua78 — APPROVED** [review]

**Pull request merged**